src/middleware/auth_middleware.go

```diff
@@ -0,0 +1,84 @@
+package middleware
+
+import (
+	"fmt"
+	"github.com/dana-team/platform-backend/src/auth"
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// TokenAuthMiddleware validates the Authorization header and sets up Kubernetes client.
+func TokenAuthMiddleware(tokenProvider auth.TokenProvider) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		token, err := validateToken(c)
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"error": err.Error()})
+			return
+		}
+
+		username, err := tokenProvider.ObtainUsername(token)
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": "Failed to get user info"})
+			return
+		}
+
+		client, err := createKubernetesClient(token)
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": "Failed to create Kubernetes client"})
+			return
+		}
+
+		// Update the logger with the username
+		if logger, exists := c.Get("logger"); exists {
+			userLogger := logger.(*zap.Logger).With(zap.String("user", username))
+			c.Set("logger", userLogger)
+		}
+
+		c.Set("kubeClient", client)
+		c.Next()
+	}
+}
+
+// validateToken validates the format and presence of the Authorization token.
+func validateToken(c *gin.Context) (string, error) {
+	token := c.GetHeader("Authorization")
+	if token == "" {
+		return "", fmt.Errorf("authorization token not provided")
+	}
+
+	tokenParts := strings.Split(token, " ")
+	if len(tokenParts) != 2 || tokenParts[0] != "Bearer" {
+		return "", fmt.Errorf("invalid authentication token")
+	}
+
+	return tokenParts[1], nil
+}
+
+// createKubernetesClient creates a Kubernetes client using the provided token.
+func createKubernetesClient(token string) (*kubernetes.Clientset, error) {
+	skipTlsVerify, err := strconv.ParseBool(os.Getenv("INSECURE_SKIP_VERIFY"))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse INSECURE_SKIP_VERIFY: %v", err)
+	}
+
+	config := &rest.Config{
+		BearerToken: token,
+		Host:        os.Getenv("KUBE_API_SERVER"),
```

Comment:
move the host as parameter to this function to prepare for the multi cluster in the architecture